scheduled_messages: Display time for messages scheduled within 5 mins.

akarsh-jain-790 opened the pull request:
Fixes: #26784.

<!-- Describe your pull request here.-->

Fixes: <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

https://github.com/zulip/zulip/assets/72064462/5f8f6f3d-2890-4e48-b2cc-760ab01b94b1

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [X] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [X] Explains differences from previous plans (e.g., issue description).
- [X] Highlights technical choices and bugs encountered.
- [X] Calls out remaining decisions and concerns.
- [X] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [X] Each commit is a coherent idea.
- [X] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [X] Visual appearance of the changes.
- [X] Responsiveness and internationalization.
- [X] Strings and tooltips.
- [X] End-to-end functionality of buttons, interactions and flows.
- [X] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
@akarsh-jain-790 can you explain your understanding of the bug and why this is the correct way to fix it? I see that it's corrected the reproducer, but I don't understand whether this might cause other problems.

akarsh-jain-790 commented:
> @akarsh-jain-790 can you explain your understanding of the bug and why this is the correct way to fix it? I see that it's corrected the reproducer, but I don't understand whether this might cause other problems.

The bug was related to the behavior of displaying the scheduled time for messages scheduled within 5 minutes in the future. The code for calculating and displaying this time relied on the `get_formatted_selected_send_later_time` function, which was used in two places.

Previously, when rendering `render_send_later_popover` there was a check that if `selected_send_later_timestamp` was less than `MINIMUM_SCHEDULED_MESSAGE_DELAY_SECONDS`, it would return undefined, effectively not displaying the scheduled time. This check was intended to handle cases where the message was considered "expired." This check caused the issue when the same function was called after scheduling a message that was less than 5 minutes in the future. The function returned undefined, and the time was not shown in the banner.

To fix this I removed the check for expiration because `selected_send_later_timestamp` was already set to be 5 minutes in the future when a message was scheduled. By removing this check, we ensured that the actual scheduled time would be displayed in the banner as expected. And added a condition to check for the `render_send_later_popover`.

alya commented:
Thanks for the work here! Closing in favor of https://github.com/zulip/zulip/pull/27642.
